source/common/config/grpc_mux_impl.h

```diff
@@ -96,6 +98,20 @@ class GrpcMuxImpl : public GrpcMux,
     TokenBucketPtr limit_request_;
     // Limits warning messages when too many requests is detected.
     TokenBucketPtr limit_log_;
+    // BackOffStrategy used for update rejected cases.
+    BackOffStrategyPtr rejected_backoff_strategy_;
+    // Timer used for update rejected cases.
+    Event::TimerPtr rejected_timer_;
+
+    void enableRejectedTimer() {
+      rejected_timer_->enableTimer(
+          std::chrono::milliseconds(rejected_backoff_strategy_->nextBackOffMs()));
+    }
+
+    void resetRejectedState() {
+      rejected_backoff_strategy_->reset();
```

Comment:
Where does this get recreated?

Reply:
This just resets the retry count to 1 and next time when a failure happens for this typeUrl, it will start from fresh back-off start value.